From 0257129a943a339c16454026a1d244f29557d545 Mon Sep 17 00:00:00 2001
From: Dennis Witzig <dennis@wtzg.de>
Date: Tue, 19 May 2020 09:53:32 +0200
Subject: [PATCH] Add vanitygaps support for fibonacci layout

---
 fibonacci.c | 82 +++++++++++++++++++++++++++--------------------------
 1 file changed, 42 insertions(+), 40 deletions(-)

diff --git a/fibonacci.c b/fibonacci.c
index fce0a57..37f7539 100644
--- a/fibonacci.c
+++ b/fibonacci.c
@@ -1,54 +1,54 @@
 void
-fibonacci(Monitor *mon, int s) {
+fibonacci(Monitor *mon, int s)
+{
 	unsigned int i, n, nx, ny, nw, nh;
 	Client *c;
 
-	for(n = 0, c = nexttiled(mon->clients); c; c = nexttiled(c->next), n++);
-	if(n == 0)
+	for (n = 0, c = nexttiled(mon->clients); c; c = nexttiled(c->next), n++);
+	if (n == 0)
 		return;
-	
-	nx = mon->wx;
-	ny = 0;
-	nw = mon->ww;
-	nh = mon->wh;
-	
-	for(i = 0, c = nexttiled(mon->clients); c; c = nexttiled(c->next)) {
-		if((i % 2 && nh / 2 > 2 * c->bw)
+
+	nx = mon->wx + mon->gappov;
+	ny = mon->gappoh;
+	nw = mon->ww - 2*mon->gappov;
+	nh = mon->wh - 2*mon->gappoh;
+
+	for (i = 0, c = nexttiled(mon->clients); c; c = nexttiled(c->next)) {
+		if ((i % 2 && nh / 2 > 2 * c->bw)
 		   || (!(i % 2) && nw / 2 > 2 * c->bw)) {
-			if(i < n - 1) {
-				if(i % 2)
-					nh /= 2;
+			if (i < n - 1) {
+				if (i % 2)
+					nh = (nh - mon->gappih) / 2;
 				else
-					nw /= 2;
-				if((i % 4) == 2 && !s)
-					nx += nw;
-				else if((i % 4) == 3 && !s)
-					ny += nh;
+					nw = (nw - mon->gappiv) / 2;
+				if ((i % 4) == 2 && !s)
+					nx += nw + mon->gappiv;
+				else if ((i % 4) == 3 && !s)
+					ny += nh + mon->gappih;
 			}
-			if((i % 4) == 0) {
-				if(s)
-					ny += nh;
+			if ((i % 4) == 0) {
+				if (s)
+					ny += nh + mon->gappih;
 				else
-					ny -= nh;
+					ny -= nh + mon->gappih;
 			}
-			else if((i % 4) == 1)
-				nx += nw;
-			else if((i % 4) == 2)
-				ny += nh;
-			else if((i % 4) == 3) {
-				if(s)
-					nx += nw;
+			else if ((i % 4) == 1)
+				nx += nw + mon->gappiv;
+			else if ((i % 4) == 2)
+				ny += nh + mon->gappih;
+			else if ((i % 4) == 3) {
+				if (s)
+					nx += nw + mon->gappiv;
 				else
-					nx -= nw;
+					nx -= nw + mon->gappiv;
 			}
-			if(i == 0)
-			{
-				if(n != 1)
-					nw = mon->ww * mon->mfact;
-				ny = mon->wy;
+			if (i == 0) {
+				if (n != 1)
+					nw = (mon->ww - 2*mon->gappov - mon->gappiv) * mon->mfact;
+				ny = mon->wy + mon->gappoh;
 			}
-			else if(i == 1)
-				nw = mon->ww - nw;
+			else if (i == 1)
+				nw = mon->ww - nw - mon->gappiv - 2*mon->gappov;
 			i++;
 		}
 		resize(c, nx, ny, nw - 2 * c->bw, nh - 2 * c->bw, False);
@@ -56,11 +56,13 @@ fibonacci(Monitor *mon, int s) {
 }
 
 void
-dwindle(Monitor *mon) {
+dwindle(Monitor *mon)
+{
 	fibonacci(mon, 1);
 }
 
 void
-spiral(Monitor *mon) {
+spiral(Monitor *mon)
+{
 	fibonacci(mon, 0);
 }
-- 
2.26.2

